services/frontend-service/src/ui/components/ProductVersion/ProductVersion.test.tsx

```diff
@@ -146,3 +146,89 @@ describe('Product Version Data', () => {
         });
     });
 });
+
+describe('Test table filtering', () => {
+    type TestData = {
+        name: string;
+        productSummary: ProductSummary[];
+        teams: string[];
+    };
+    const data: TestData[] = [
+        {
+            name: 'no rows to display',
+            productSummary: [],
+            teams: [],
+        },
+        {
+            name: 'no teams to filter out',
+            productSummary: [
+                {
+                    app: 'testing-app',
+                    version: '4',
+                    commitId: '123',
+                    displayVersion: 'v1.2.3',
+                    environment: 'dev',
+                    team: 'sre-team',
+                },
+            ],
+            teams: [],
+        },
+        {
+            name: 'a team to filter out',
+            productSummary: [
+                {
+                    app: 'testing-app',
+                    version: '4',
+                    commitId: '123',
+                    displayVersion: 'v1.2.3',
+                    environment: 'dev',
+                    team: 'sre-team',
+                },
+                {
+                    app: 'testing-app',
+                    version: '4',
+                    commitId: '123',
+                    displayVersion: 'v1.2.3',
+                    environment: 'dev',
+                    team: 'others',
+                },
+            ],
+            teams: ['sre-team'],
+        },
+        {
+            name: 'a team to filter out',
+            productSummary: [
+                {
+                    app: 'testing-app',
+                    version: '4',
+                    commitId: '123',
+                    displayVersion: 'v1.2.3',
+                    environment: 'dev',
+                    team: 'sre-team',
+                },
+                {
+                    app: 'testing-app 2',
+                    version: '4',
+                    commitId: '123',
+                    displayVersion: 'v1.2.3',
+                    environment: 'dev',
+                    team: 'others',
+                },
+                {
+                    app: 'testing-app 3',
+                    version: '4',
+                    commitId: '123',
+                    displayVersion: 'v1.2.3',
+                    environment: 'dev',
+                    team: 'others',
+                },
+            ],
+            teams: ['sre-team', 'others'],
+        },
+    ];
+    describe.each(data)(`Displays Product Version Table`, (testCase) => {
+        render(<TableFiltered productSummary={testCase.productSummary} teams={testCase.teams} />);
+        expect(document.querySelector('.table')?.textContent).toContain('App Name');
+        expect(document.body).toMatchSnapshot();
```

Comment:
please remove "toMatchSnapshot" and replace it with individual "expect" invocations.